### ISSUE TITLE
Deadlock with database in Git2DB

### ISSUE THREAD

**Issue by jlcanovas** [SOM-Research/Gitana#8]
While extracting data from a set of git repositories the Git2DB process triggers this error related to the connection with the database:

```
2016-10-05 18:54:10:ERROR:Git2Db failed
Traceback (most recent call last):
  File "/home/recerca/Gitana/extractor/cvs/git/git2db_extract_reference.py", line 468, in extract
    self.get_info_contribution_in_reference(self.ref_name, self.repo_id, self.from_sha)
  File "/home/recerca/Gitana/extractor/cvs/git/git2db_extract_reference.py", line 299, in get_info_contribution_in_reference
    self.analyse_commits(commits, reference_name, repo_id)
  File "/home/recerca/Gitana/extractor/cvs/git/git2db_extract_reference.py", line 443, in analyse_commits
    self.analyse_commit(c, ref_id, repo_id)
  File "/home/recerca/Gitana/extractor/cvs/git/git2db_extract_reference.py", line 434, in analyse_commit
    self.insert_commit_parents(commit.parents, commit_found, sha, repo_id)
  File "/home/recerca/Gitana/extractor/cvs/git/git2db_extract_reference.py", line 168, in insert_commit_parents
    cursor.execute(query, arguments)
  File "/usr/lib/python2.7/site-packages/mysql/connector/cursor.py", line 515, in execute
    self._handle_result(self._connection.cmd_query(stmt))
  File "/usr/lib/python2.7/site-packages/mysql/connector/connection.py", line 488, in cmd_query
    result = self._handle_result(self._send_cmd(ServerCmd.QUERY, query))
  File "/usr/lib/python2.7/site-packages/mysql/connector/connection.py", line 395, in _handle_result
    raise errors.get_exception(packet)
InternalError: 1213 (40001): Deadlock found when trying to get lock; try restarting transaction
```

In particular, this issue appeared when working with `docker/docker` GitHub repository

**Comment by valeriocos** [comment]
the problem may appear when two (or more) processes Git2DbReference try to access/modify the table commit_parents. It should be fixed in the next commit

**Comment by valeriocos** [comment]
fixed in commit 38394deeb0f33e8b299288887ea565e0834b0371
